pkg/async/schedule/flytescheduler/go_gf.go

```diff
@@ -0,0 +1,137 @@
+package flytescheduler
+
+import (
+	"context"
+	"fmt"
+	"github.com/flyteorg/flyteadmin/pkg/async/schedule/aws"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/gogf/gf/os/gcron"
+	"github.com/gogf/gf/os/gtimer"
+	"github.com/robfig/cron"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type GoGF struct {
+}
+
+func (g GoGF) GetScheduleName(s models.SchedulableEntity) string {
+	return strconv.FormatUint(aws.HashIdentifier(core.Identifier{
+		Project: s.Project,
+		Domain:  s.Domain,
+		Name:    s.Name,
+		Version: s.Version,
+	}), 10)
+}
+
+func (g GoGF) Register(ctx context.Context, s models.SchedulableEntity, funcRef func()) error {
+	nameOfSchedule := g.GetScheduleName(s)
+
+	if s.Active {
+		if len(s.CronExpression) > 0 {
+			err := addCronJob(s.CronExpression,funcRef,nameOfSchedule)
+			if err != nil {
+				logger.Errorf(ctx, "failed to add cron schedule %v due to %v", s, err)
+			}
+		} else {
+			err := addFixedIntervalJob(s.Unit, s.FixedRateValue, funcRef)
+			if err != nil {
+				logger.Errorf(ctx, "failed to add fixed rate schedule %v due to %v", s, err)
+			}
+		}
+	} else {
+		if len(s.CronExpression) > 0 {
+			removeCronJob(ctx, nameOfSchedule)
```

Comment:
Register from GoGF perspective is any new schedule that it receives which can either be an activation in which case it will add or a deactivation in which case it will remove.

Do we want to name this something else ?

Reply:
got it, maybe a comment to explain?